File: kotlin/darts/src/main/kotlin/Darts.kt
object Darts {

    fun score(x: Number, y: Number): Int {
        val xd = x.toDouble();
        val yd = y.toDouble();
        val r_sq = xd * xd + yd * yd;
        if (r_sq <= 1) return 10
        if (r_sq <= 25) return 5
        if (r_sq <= 100) return 1
        return 0
    }
}

File: free-pascal/darts/1/Darts.pas
unit Darts;

{$mode ObjFPC}{$H+}

interface

function score(const x : single; const y : single) : uint8;

implementation

uses SysUtils;

var r_sq : real;

function score(const x : single; const y : single) : uint8;
begin

  r_sq := x*x + y*y;
  if r_sq <= 1 then Exit(10);
  if r_sq <= 25 then Exit(5);
  if r_sq <= 100 then Exit(1);
  result := 0;

end;

end.
